src/Tasks/dependency-check-build-task/dependency-check-build-task.ts

```diff
@@ -176,13 +209,41 @@ async function run() {
                 console.log(`##vso[build.uploadlog]${logFile}`);
         }
 
+        let message = "Dependency Check succeeded"
+        let result = tl.TaskResult.Succeeded
         if (failed) {
-            let message = "Dependency Check exited with an error code.";
-            if (isViolation) message = "CVSS threshold violation.";
+            if(isViolation) {
+                message = "CVSS threshold violation.";
+
+                if(warnOnCVSSViolation) {
+                    result = tl.TaskResult.SucceededWithIssues
+                }
+                else {
+                    result = tl.TaskResult.Failed
+                }
+            }
+            else {
+                message = "Dependency Check exited with an error code (exit code: " + exitCode + ")."
```

Comment:
Fail build step if dependency check tool exited with any code other than 0 or 1 (violation) and print out the exit code.